src/main/java/seedu/address/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,121 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Comparator;
+import java.util.List;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+public class SortCommand extends Command {
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorts entries in the AddressBook by the specified field(s).\n"
+            + "Order parameters are mandatory but specifying order is optional.\n"
+            + "All other parameters are optional.\n"
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_PHONE + "PHONE] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_STATUS + "STATUS] "
+            + "[" + PREFIX_TAG + "TAG] "
+            + "[" + PREFIX_ORDER + "ORDER]\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_TAG + " " + PREFIX_NAME
+            + " " + PREFIX_ORDER + "desc" + "\n";
+
+    public static final String MESSAGE_SUCCESS = "Sorted Modules successfully";
+
+    private final PersonComparator personComparator;
+    private final List<Prefix> fields;
+    /**
+     * @param fields modules to be deleted
+     */
+    public SortCommand(List<Prefix> fields, String order) {
+        this.fields = fields;
+        this.personComparator = new PersonComparator(fields, order);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.sortPerson(personComparator);
+        return new CommandResult(MESSAGE_SUCCESS + ": " + fields.toString());
+    }
+
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        return (other instanceof SortCommand)
+                && this.fields.equals(((SortCommand) other).fields)
+                && this.personComparator.equals(((SortCommand) other).personComparator); // instanceof handles null
+    }
+
+    public static class PersonComparator implements Comparator<Person> {
+        private final List<Prefix> fields;
+        private final String order;
+        /**
+         * Create a comparator using the specified fields,
+         * using the ordering implied by its iterator.
+         * @param fields a list of field names
+         */
+        public PersonComparator(List<Prefix> fields, String order) {
+            this.fields = fields;
+            this.order = order;
+        }
+
+        @Override
+        public int compare(Person o1, Person o2) {
+            for (Prefix field : fields) {
+                int result = 0;
+                if (PREFIX_NAME.equals(field)) {
```

Comment:
Minor, but may look cleaner if we use `switch case` statements here instead.

Reply:
Cannot use Prefix class for ``switch`` statement.
>The type of the Expression must be char, byte, short, int, Character, Byte, Short, Integer, String, or an enum type (§8.9), or a compile-time error occurs.

https://stackoverflow.com/questions/22185706/using-java-lang-class-in-a-switch-statement